src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -39,30 +36,28 @@ function SaveResponseInOnyx(response, request) {
             return true;
         });
 
-        // Save the update IDs to Onyx so they can be used to fetch incremental updates if the client gets out of sync from the server
-        OnyxUpdates.saveUpdateIDs(Number(responseData.lastUpdateID || 0), Number(responseData.previousUpdateID || 0));
+        const responseToApply = {
+            type: CONST.ONYX_UPDATE_TYPES.HTTPS,
+            lastUpdateID: Number(responseData.lastUpdateID || 0),
+            data: {
+                request,
+                responseData,
+            },
+        };
 
-        // For most requests we can immediately update Onyx. For write requests we queue the updates and apply them after the sequential queue has flushed to prevent a replay effect in
-        // the UI. See https://github.com/Expensify/App/issues/12775 for more info.
-        const updateHandler = request.data.apiRequestType === CONST.API_REQUEST_TYPE.WRITE ? QueuedOnyxUpdates.queueOnyxUpdates : Onyx.update;
+        if (_.includes(requestsToIgnoreLastUpdateID, request.command) || !OnyxUpdates.needsToUpdateClient(Number(responseData.previousUpdateID || 0))) {
+            return OnyxUpdates.apply(responseToApply);
+        }
 
-        // First apply any onyx data updates that are being sent back from the API. We wait for this to complete and then
-        // apply successData or failureData. This ensures that we do not update any pending, loading, or other UI states contained
-        // in successData/failureData until after the component has received and API data.
-        const onyxDataUpdatePromise = responseData.onyxData ? updateHandler(responseData.onyxData) : Promise.resolve();
+        // Save the update IDs to Onyx so they can be used to fetch incremental updates if the client gets out of sync from the server
+        OnyxUpdates.saveUpdateInformation(responseToApply, Number(responseData.lastUpdateID || 0), Number(responseData.previousUpdateID || 0));
 
-        return onyxDataUpdatePromise
-            .then(() => {
-                // Handle the request's success/failure data (client-side data)
-                if (responseData.jsonCode === 200 && request.successData) {
-                    return updateHandler(request.successData);
-                }
-                if (responseData.jsonCode !== 200 && request.failureData) {
-                    return updateHandler(request.failureData);
-                }
-                return Promise.resolve();
-            })
-            .then(() => responseData);
+        // We'll add the pauseQueue property here to guarantee we're pausing the queue before we execute the next request. If we do it only on the trigger in
+        // OnyxUpdateManager, then we might have already started another request while we wait.
```

Comment:
```suggestion
        // Ensure the queue is paused while the client resolves the gap in onyx updates so that updates are guaranteed to happen in a specific order. 
```

Reply:
Done